packages/medusa/src/api/routes/admin/order-edits/request-confirmation.ts

```diff
@@ -54,26 +60,65 @@ import {
  */
 export default async (req, res) => {
   const { id } = req.params
+  const validatedBody =
+    req.validatedBody as AdminPostOrderEditsRequestConfirmationReq
 
   const orderEditService: OrderEditService =
     req.scope.resolve("orderEditService")
 
+  const paymentCollectionService: PaymentCollectionService = req.scope.resolve(
+    "paymentCollectionService"
+  )
+
   const manager: EntityManager = req.scope.resolve("manager")
 
   const loggedInUser = (req.user?.id ?? req.user?.userId) as string
 
   await manager.transaction(async (transactionManager) => {
-    await orderEditService
-      .withTransaction(transactionManager)
-      .requestConfirmation(id, { loggedInUserId: loggedInUser })
+    const orderEditServiceTx =
+      orderEditService.withTransaction(transactionManager)
+
+    await orderEditServiceTx.requestConfirmation(id, {
+      loggedInUserId: loggedInUser,
+    })
+    const total = await orderEditServiceTx.getTotals(orderEdit.id)
+
+    if (total.difference_due > 0) {
+      const paymentCollectionServiceTx =
+        paymentCollectionService.withTransaction(transactionManager)
+
+      const orderEditRepo: OrderEditRepository =
+        transactionManager.getCustomRepository(
+          req.scope.resolve("orderEditRepository")
+        )
+
+      const paymentCollection = await paymentCollectionServiceTx.create({
+        type: PaymentCollectionType.ORDER_EDIT,
+        amount: total.difference_due,
+        currency_code: orderEdit.order.currency_code,
+        region_id: orderEdit.order.region_id,
+        description: validatedBody.payment_collection_description,
+        created_by: loggedInUser,
+      })
+
+      orderEdit.payment_collection_id = paymentCollection.id
+      await orderEditRepo.save(orderEdit)
+    }
   })
 
-  const orderEdit = await orderEditService.retrieve(id, {
+  let orderEdit = await orderEditService.retrieve(id, {
     relations: defaultOrderEditRelations,
     select: defaultOrderEditFields,
   })
+  orderEdit = await orderEditService.decorateTotals(orderEdit)
 
   res.status(200).send({
     order_edit: orderEdit,
   })
 }
+
+export class AdminPostOrderEditsRequestConfirmationReq {
+  @IsString()
+  @IsOptional()
+  payment_collection_description?: string | undefined
+}
```

Comment:
question: what is the point of the description?

Reply:
I guess we agreed that a Payment Collection could have a description. That's the only reason